datajunction-server/datajunction_server/sql/dag.py

```diff
@@ -882,6 +882,8 @@ async def get_nodes_with_dimension(
     final_set: Set[NodeRevision] = set()
     while to_process:
         current_node = to_process.pop()
+        if current_node.name in processed:
+            continue
         processed.add(current_node.name)
```

Comment:
Wondering why you add the node to `processed` set before it is actually processing in the lines below. Or if you actually want to do that then perhaps this variable should be really called `processing_started_for` ?